src/garage/envs/gym_env.py

```diff
@@ -0,0 +1,388 @@
+        This method can be removed once OpenAI solves the issue.
+        """
+        # We need to do some strange things here to fix-up flaws in gym
+        # pylint: disable=import-outside-toplevel
+        if self._env.spec:
+            if any(package in getattr(self._env.spec, 'entry_point', '')
+                   for package in KNOWN_GYM_NOT_CLOSE_MJ_VIEWER):
+                # This import is not in the header to avoid a MuJoCo dependency
+                # with non-MuJoCo environments that use this base class.
+                try:
+                    from mujoco_py.mjviewer import MjViewer
+                    import glfw
+                except ImportError:
+                    # If we can't import mujoco_py, we must not have an
+                    # instance of a class that we know how to close here.
+                    return
+                if (hasattr(self._env, 'viewer')
+                        and isinstance(self._env.viewer, MjViewer)):
+                    glfw.destroy_window(self._env.viewer.window)
+            elif any(package in getattr(self._env.spec, 'entry_point', '')
+                     for package in KNOWN_GYM_NOT_CLOSE_VIEWER):
+                if hasattr(self._env, 'viewer'):
+                    from gym.envs.classic_control.rendering import (
+                        Viewer, SimpleImageViewer)
+                    if (isinstance(self._env.viewer,
+                                   (SimpleImageViewer, Viewer))):
+                        self._env.viewer.close()
+
+    def __getstate__(self):
+        """See `Object.__getstate__.
+
+        Returns:
+            dict: The instance’s dictionary to be pickled.
+
+        """
+        # the viewer object is not pickleable
+        # we first make a copy of the viewer
+        env = self._env
+
+        # get the inner env if it is a gym.Wrapper
+        if issubclass(env.__class__, gym.Wrapper):
+            env = env.unwrapped
+
+        if 'viewer' in env.__dict__:
+            _viewer = env.viewer
+            # remove the viewer and make a copy of the state
+            env.viewer = None
+            state = copy.deepcopy(self.__dict__)
+            # assign the viewer back to self.__dict__
+            env.viewer = _viewer
+            # the returned state doesn't have the viewer
+            return state
+        return self.__dict__
+
+    def __setstate__(self, state):
+        """See `Object.__setstate__.
+
+        Args:
+            state (dict): Unpickled state of this object.
+
+        """
+        self.__init__(state['_env'])
+
+    def __getattr__(self, attr):
+        """Handle function calls wrapped environment.
+
+        Args:
+            attr (str): attribute name
+
+        Returns:
+            object: the wrapped attribute, or the return value of
+                wrapped function.
+
+        Raises:
+            AttributeError: if the attribute is not found in the wrapped env.
+        """
+        if hasattr(self._env, attr):
```

Comment:
does this break `help(obj)` and/or `dir(obj)`? If so, please find another way.

Reply:
wait -- why would i pass through attributes to the wrapped class? that allows me to casually violate the interface.

if you want to expose the wrapped instance, perhaps add some sort of property? `GymEnv.unwrapped`?